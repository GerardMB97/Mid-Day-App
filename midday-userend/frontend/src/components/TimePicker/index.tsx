import React, { useState } from 'react';
import { TouchableWithoutFeedback, View, StyleSheet } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import DateTimePickerModal from 'react-native-modal-datetime-picker';

interface DatePickerProps{
  selectedHour: string,
  setSelectedHour: Function
}

const styles = StyleSheet.create({
  icon: {
    fontSize: 30
  }
});

const DatePicker = ({ selectedHour, setSelectedHour }:DatePickerProps) => {
  const [isDatePickerVisible, setDatePickerVisibility] = useState(false);

  const handleConfirm = (date : Date) => {
    const hour = date.getHours().toString();
    const minutes = date.getMinutes().toString().split('');
    const modifiedMinutes = minutes[0] + '0';
    const selectedHour = `${hour}:${modifiedMinutes}`;

    return selectedHour;
  };

  return (
      <TouchableWithoutFeedback testID="show" onPress={() => setDatePickerVisibility(true)} >
        <View>
        <Icon name="time-outline" style={styles.icon}></Icon>
      <DateTimePickerModal
        testID="picker"
        isVisible={isDatePickerVisible}
        minuteInterval= {30}
        headerTextIOS= 'Estamos disponibles de 12:00 a 16:00 '
        mode="time"
        locale="en_GB"
        onConfirm={(date) => setSelectedHour(handleConfirm(date))}
        onCancel={() => setDatePickerVisibility(false)}
      />
      </View>
      </TouchableWithoutFeedback>

  );
};

export default DatePicker;
