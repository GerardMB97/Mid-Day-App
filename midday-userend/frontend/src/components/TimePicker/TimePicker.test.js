import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import TimePicker from '.';

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');
jest.mock('react-native-modal-datetime-picker', () => 'Picker');

describe('Given a TimePicker component', () => {
  test('Then it should render correctly', () => {
    const rendered = render(<TimePicker></TimePicker>);

    const button = rendered.getByTestId('show');
    fireEvent.press(button);

    expect(rendered).toMatchSnapshot();
  });
});
