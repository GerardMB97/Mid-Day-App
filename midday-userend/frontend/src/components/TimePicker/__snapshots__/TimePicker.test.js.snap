// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a TimePicker component Then it should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="show"
>
  <Icon
    name="time-outline"
    style={
      Object {
        "fontSize": 30,
      }
    }
  />
  <Picker
    headerTextIOS="Estamos disponibles de 12:00 a 16:00 "
    isVisible={true}
    locale="en_GB"
    minuteInterval={30}
    mode="time"
    onCancel={[Function]}
    onConfirm={[Function]}
    testID="picker"
  />
</View>
`;
